### PR TITLE
Feat(demo)/simple open auction

### PR DESCRIPTION
**Pull request by vuvoth**
### Proposal
Add simple open action demo:
- [Fe #499](https://github.com/ethereum/fe/issues/499)
- [simple open action solidity](https://docs.soliditylang.org/en/latest/solidity-by-example.html#simple-open-auction)

### Implementation
- Add demo contract simple_open_auction.fe.
- Happy test cases for bid method.

### To-Do
- [x] Unhappy test case for bid. 
- [x] withdraw test cases.
- [x] action_end test cases.

[//]: # (Stay ahead of things, add list items here!)
- [ ] OPTIONAL: Update [Spec](https://github.com/ethereum/fe/blob/master/docs/src/spec/index.md) if applicable
- [ ] Add entry to the [release notes](https://github.com/ethereum/fe/blob/master/newsfragments/README.md) (may forgo for trivial changes)

- [ ] Clean up commit history

### CODE REVIEW

File: crates/test-files/fixtures/demos/simple_open_auction.fe [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    
+    # constructor 
+    pub fn __init__(self, ctx: Context, bidding_time: u256, beneficiary_addr: address): 
+        self.beneficiary = beneficiary_addr
+        self.auction_end_time = ctx.block_timestamp() + bidding_time
+    
+    #method
+    pub fn bid(self, ctx: Context):
+        assert ctx.block_timestamp() <= self.auction_end_time, "Auction already ended"
```
Comment (cburgdorf) [inline crates/test-files/fixtures/demos/simple_open_auction.fe:32]: What do you think about encouraging the use of custom errors here.

E.g.

```
struct AlreadyEnded:
    pass
```

and then:

```
ctx.block_timestamp() > self.auction_end_time:
    revert AlreadyEnded()
```

**Review by cburgdorf — COMMENTED** [review]
Great work! Left a comment inline for a small potential improvement

File: crates/test-files/fixtures/demos/simple_open_auction.fe [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        assert ctx.msg_value() > self.highest_bid, "Bid not high enough"
```
[the remark this answers is not in the document]
Comment (vuvoth) [reply crates/test-files/fixtures/demos/simple_open_auction.fe:34]: Thank FYI ! I didn't we supported revert with struct(error).

**Comment by vuvoth** [comment]
> Great work! Left a comment inline for a small potential improvement

I pushed new code. Please check it, thanks!

**Review by cburgdorf — APPROVED** [review]

**Comment by cburgdorf** [comment]
Thanks again @vuvoth :tada:

**Pull request merged**